dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java

```diff
@@ -572,4 +573,83 @@ public void testLoadCategories_GivenContentWithCats_ShouldLoadESMapWithListOfCat
 
     }
 
+    /**
+     * General purpose is testing that KeyValue fields are indexed correctly.
+     * Given scenario: We create a Content type that holds a Key value field.
+     * Expected Results: Then we feed data into such field and use an ES query over the KeyValue to verify the results are coming back.
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void Test_Create_ContentType_With_KeyValue_Field_Test_Query_Expect_Success()
+            throws DotDataException, DotSecurityException {
+
+        List<Field> fields = new ArrayList<>();
+        final String myKeyValueField = "myKeyValueField";
+        fields.add(
+                new FieldDataGen()
+                        .type(KeyValueField.class)
+                        .name(myKeyValueField)
+                        .velocityVarName(myKeyValueField).indexed(true)
+                        .next()
+        );
+        final String contentTypeName = "myContentTypeWithKeyValues" + System.currentTimeMillis();
+        final ContentType contentType = new ContentTypeDataGen()
+                .name(contentTypeName)
+                .velocityVarName(contentTypeName)
+                .fields(fields)
+                .nextPersisted();
+
+        new ContentletDataGen(contentType.id()).setProperty(myKeyValueField,
+                "{\"key\":\"key1\", value:\"val\" }").nextPersisted();
+        final String queryString =  String.format("+%s.%s.key:%s*",contentTypeName, myKeyValueField, "val");
+        Logger.info(ESMappingAPITest.class, () -> String.format(" Query: %s ",queryString));
+        final String wrappedQuery = String.format("{"
+                + "query: {"
+                + "   query_string: {"
+                + "        query: \"%s\""
+                + "     }"
+                + "  }"
+                + "}", queryString);
+
+        final ESSearchResults searchResults = contentletAPI.esSearch(wrappedQuery, false,  user, false);
+        Assert.assertFalse(searchResults.isEmpty());
+        for (Object searchResult : searchResults) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'searchResult' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5533307)